packages/dd-trace/src/dogstatsd.js

```diff
@@ -11,6 +13,16 @@ class Client {
   constructor (options) {
     options = options || {}
 
+    if (options.tracingUrl) {
+      this._httpOptions = url.parse(options.tracingUrl)
+      this._httpOptions.agent = null
+      if (this._httpOptions.protocol === 'unix:') {
```

Comment:
Why is this needed? It should be handled by the underlying HTTP client by moving the code from the writer.

Reply:
We do _some_ normalization today in `request`, but it takes in an options object, rather than a URL. I _guess_ we could put this into that normalization somehow since it's also happening in writer, but that would mean an extra property on the options object I think. Maybe `url`. I'll give that a try.